twisted/conch/ssh/keys.py

```diff
@@ -802,9 +817,21 @@ def fingerprint(self):
 
         @rtype: L{str}
         """
-        return nativeString(
-            b':'.join([binascii.hexlify(x)
-                       for x in iterbytes(md5(self.blob()).digest())]))
+        hashFunction = {'md5-hex' : md5, 'sha256-base64' : sha256}
+
+        try:
+            fingerprint = hashFunction[format](
+                self.blob()).digest()
+        except KeyError:
+            raise BadFingerPrintFormat(
+                "unsupported fingerprint format: %s" % format)
+
+        if format == 'sha256-base64':
+            return nativeString(base64.b64encode(fingerprint))
+        else:
+            return nativeString(
```

Comment:
this is wrong... what would happen if I ask for 'md5-bogus'  of 'sha256-bogus' ? ... it looks like it will still use hex ... and I think that is wrong as instead we should have an error